algorithm/MovieLens.R

```diff
@@ -10,48 +11,49 @@ library(sampling)
 #
 # Datasets are acquired from website https://grouplens.org/datasets/movielens
 setwd("Documents/R/datasets/ml-100k")
-data <- read.table("u.data")[ , -4]
-names(data) <- c("UserID", "MovieID", "Rating")
+rating <- read.table("u.data")[ , -4]
+names(rating) <- c("UserID", "MovieID", "Rating")
 
 # Compile the function "MovieLensKNN"
-MovieLensKNN <- function(userID, movieID, n, K) {
-  targetUsers <- which(data$UserID == userID)
-  relativeMovies <- sample(targetUsers, min(length(targetUsers), n))
+MovieLensKNN <- function(userID, movieID, n, k) {
+  targetUserIndexes <- which(rating$UserID == userID)
+  sampleIndexes <- sample(targetUserIndexes, min(length(targetUserIndexes), n))
 
-  relativeMoviesID <- data$MovieID[relativeMovies]
-  targetMoviesID <- movieID
+  sampleMovieIDs <- rating$MovieID[sampleIndexes]
+  targetMovieID <- movieID
 
-  trainingMovies <- which(data$MovieID == targetMoviesID)
-  trainingUsers <- data$UserID[trainingMovies[-1]]
+  targetMovieIndexes <- which(rating$MovieID == targetMovieID)
+  targetUser <- which(rating$UserID[targetMovieIndexes] == userID)
+  trainingUsers <- rating$UserID[targetMovieIndexes][-targetUser]
 
-  dataAll <- matrix(0, 1+length(trainingUsers), 2+length(relativeMoviesID))
-  dataAll <- data.frame(dataAll)
-  names(dataAll) <- c("userID",
+  dataMatrix <- matrix(0, 1+length(trainingUsers), 2+length(sampleMovieIDs))
+  dataDF <- data.frame(dataMatrix)
+  names(dataDF) <- c("userID",
                       paste("targetMovieID-", movieID),
-                      paste("tuNotedMovieID-", relativeMoviesID, sep = ""))
-  movies <- c(targetMoviesID, relativeMoviesID)
-  dataAll$userID <- c(userID, trainingUsers)
+                      paste("sampleMovieID-", sampleMovieIDs, sep = ""))
+  movies <- c(targetMovieID, sampleMovieIDs)
+  dataDF$userID <- c(userID, trainingUsers)
 
-  for (i in 1:nrow(dataAll)) {
-    dataUsers <- data[which(data$UserID == dataAll$userID[i]), ]
+  for (i in 1:nrow(dataDF)) {
+    dataUsers <- rating[which(rating$UserID == dataDF$userID[i]), ]
```

Comment:
Could you avoid double criteria (equality by id + index)?
See https://stackoverflow.com/questions/14205583/filtering-data-in-a-table-based-on-criteria